### DISCUSSION TITLE
New HomeAssistant "Siren" integration

### DISCUSSION THREAD

**Discussion by OnilDu39** [Koenkk/zigbee2mqtt#8412] [Devices]
HomeAssistant devs have recently added the Siren integration, which provides convenient services to manipulate siren devices in a generic manner : [https://www.home-assistant.io/integrations/siren/](https://www.home-assistant.io/integrations/siren/)

That said, and correct me if i'm wrong, I suppose that siren entities have to be created by the integration that already create the entities (is zigbee2mqtt case, MQTT integration), and since this feature is very new, I guess almost none of them support it.

Is that the job of the "MQTT" integration to handle this new feature ? If so, how zigbee2mqtt can configure a device so that MQTT recognize it is a siren ?

I think this new Siren feature is very nice and that it could be cool that zigbee2mqtt/mqtt support it so that all the supported sirens can work.

**Comment by Koenkk** [comment]
@frenck are there plans to add siren support through MQTT?

**Reply by frenck** [reply]
Dunno? Not seen any effort for that at this point.

**Comment by Nardol** [comment]
If I understand correctly, it will be implemented in the version 2022.3: https://rc.home-assistant.io/integrations/siren.mqtt/
